aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -160,9 +169,42 @@ private String cleanWarehousePath(String path) {
     }
   }
 
+  @Override
+  public org.apache.iceberg.Table loadTable(TableIdentifier identifier) {
+    if (lakeFormationEnabled) {
```

Comment:
I think this logic is not right, we cannot first check LF access and then load the table, because the table identifier might be for something like a metadata table. e.g. `db.table.snapshots`, and this call will fail to find the table.